test/acceptance/public-api-test.js

```diff
@@ -1475,5 +1474,149 @@ describe('public api', function () {
       expect(result.output).toEqual('\uFEFF<button type="button">LOL, Click me!</button>');
       expect(result.isFixed).toEqual(true);
     });
+
+    it("[.html] files with DOCTYPE using different casings won't cause errors", async function () {
+      project.setConfig();
+
+      const templates = {
+        'DOCTYPE.html': `
+<!DOCTYPE html>
+<html>
+<head>
+  <title>MyApp</title>
+</head>
+<body>
+  <!-- COMMENT -->
+  <p>Foo</p>
+</body>
+</html>`,
+        'doctype.html': `
+<!doctype html>
+<html>
+<head>
+  <title>MyApp</title>
+</head>
+<body>
+  <!-- COMMENT -->
+  <p>Foo</p>
+</body>
+</html>`,
+        'DocType.html': `
+<!DocType html>
+<html>
+<head>
+  <title>MyApp</title>
+</head>
+<body>
+  <!-- COMMENT -->
+  <p>Foo</p>
+</body>
+</html>`,
+      };
+
+      project.write({
+        app: templates,
+      });
+
+      for (const path in templates) {
+        if (Object.prototype.hasOwnProperty.call(templates, path)) {
+          let templatePath = project.path(`app/${path}`);
+          let templateContents = fs.readFileSync(templatePath, { encoding: 'utf8' });
+          let results = await linter.verify({
+            source: templateContents,
+            filePath: templatePath,
+            moduleId: templatePath.slice(0, -4),
+          });
+
+          expect(results).toEqual([]);
+        }
+      }
+    });
+
+    it('[.html] files with DOCTYPE are fixable and updated output includes original DOCTYPE if source includes it', async function () {
```

Comment:
I don't think this example will work, because the button in an .html file won't trip the `require-button-type` rule (if that's what you were going for). The `no-invalid-meta` rule could work but it doesn't have an automatic fix available.

Reply:
It works because thats the same config set in the `describe` which had the `require-button-type` set, an earlier test was already testing this. What I am testing here is if the updated output contains the original doctype. The test pass so it shows it works.